content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,381 @@
+    <p>Check console output at &QUOT;<a href='${env.BUILD_URL}'>${env.JOB_NAME} [${env.BUILD_NUMBER}]</a>&QUOT;</p>"""
+
+  // Override default values based on build status
+  if (buildStatus == 'STARTED') {
+    color = 'YELLOW'
+    colorCode = '#FFFF00'
+  } else if (buildStatus == 'SUCCESSFUL') {
+    color = 'GREEN'
+    colorCode = '#00FF00'
+  } else {
+    color = 'RED'
+    colorCode = '#FF0000'
+  }
+
+  // Send notifications
+  slackSend color: colorCode, message: summary
+
+  hipchatSend color: color, notify: true, message: summary
+
+  emailext (
+      to: 'bitwiseman@bitwiseman.com',
+      subject: subject,
+      body: details,
+      recipientProviders: [[$class: 'DevelopersRecipientProvider']]
+    )
+}
+----
+
+image::/images/post-images/2017-02-14/blueocean-notifications.png[Blue Ocean Run with Notifications, role="center", width=800]
+
+== Moving Notifications to Shared Library
+
+Our current pipeline is pretty good, we have our Declarative Pipeline sending notifications.
+However, the notification method takes up about a third of our `Jenkinsfile`.
+It is a bit of a distraction.
+Also, as Jenkins admin, I'd like to share that notification method among all my projects
+without copying it to each new project and then having maintain all the copies.
+I want to move that method to a
+link:/doc/book/pipeline/shared-libraries/[Shared Library].
+
+Shared Libraries are not specific to Declarative. They were released in their
+current form several months ago and were useful in Scripted Pipeline.
+Since Declarative is still Pipeline, we can use them here as well.
+While not every method like this one should go into a Shared Library,
+Declarative's focus on overall pipeline flow makes it much clearer what
+parts of a pipeline are candidates for moving to a shared library.
+// TODO: Might be a good idea to put a short statement of how you judge what should and shouldn't go into a Shared Library
+
+=== Setting up a Shared Library
+
+I've created a simple shared library repository for this series of posts, called
+link:https://github.com/bitwiseman/jenkins-pipeline-shared[jenkins-pipeline-shared].
+The shared library functionality has too many configuration options to cover in one post.
+I've chosen to configure this library as a "Global Pipeline Library",
+accessible from any project on my Jenkins master.
+To setup a "Global Pipeline Library", navigated to "Manage Jenkins" -> "Configure System"
+in the Jenkins web UI.
+Then, under "Global Pipeline Libraries", I've added a new library.
+I've the name `bitwiseman-shared`, pointed it at my repository,
+and set the default branch for the library to `master` (we ),
+but I'll override that in my `Jenkinsfile`.
+
+image::/images/post-images/2017-02-14/shared-library.png[Global Pipeline Library, role="center", width=800]
+
+=== Moving the Code to the Library
+
+Moving a method or class to a library involves creating a file with the name of our method,
```

Comment:
A wee bit more detail on the shared lib structure and naming would be nice, e.g.:

creating a file *in the `vars` directory* with the name of our method — in this case `vars/sendNotifications.groovy`